src/main/java/com/ita/if103java/ims/service/impl/ItemServiceImpl.java

```diff
@@ -88,46 +90,54 @@ private boolean isEnoughCapacityInWarehouse(SavedItemDto savedItemDto) {
+    public boolean softDelete(Long id) {
+        return itemDao.softDeleteItem(id);
     }
 
     @Override
-    public SavedItemDto findByItemDto(ItemDto itemDto) {
-        return savedItemDtoMapper.convertSavedItemToSavedItemDto(savedItemDao.findSavedItemById(itemDto.getId()));
+    public List<SavedItemDto> findByItemId(Long id) {
+        return savedItemDtoMapper.toDtoList(savedItemDao.findSavedItemByItemId(id));
     }
 
 
     @Override
-    public boolean moveItem(WarehouseLoadDto warehouseLoadDto, SavedItemDto savedItemDto) {
-        savedItemDto.setItemDto(itemDtoMapper.convertItemToItemDto(itemDao.findItemById(savedItemDto.getId())));
+    public boolean moveItem(SavedItemDto savedItemDto, Long id, User user, Associate associate) {
+        savedItemDto.setItemDto(itemDtoMapper.toDto(itemDao.findItemById(savedItemDto.getId())));
         if (isEnoughCapacityInWarehouse(savedItemDto)) {
-            return savedItemDao.updateSavedItem(warehouseLoadDto.getId(), savedItemDto.getId());
+            eventService.create(ItemEventUtil.createEvent(savedItemDto, user, EventName.ITEM_MOVED));
+            transactionDao.create(ItemTransactionUtil.createTransaction(savedItemDto, user, associate, TransactionType.MOVE));
+            eventService.create(ItemEventUtil.createEvent(savedItemDto, user, EventName.ITEM_MOVED));
+            return savedItemDao.updateSavedItem(id, savedItemDto.getId());
         }
+        eventService.create(ItemEventUtil.createEvent(savedItemDto, user, EventName.LOW_SPACE_IN_WAREHOUSE));
         throw new ItemNotEnoughCapacityInWarehouseException("Can't move savedItemDto in warehouse because it doesn't " +
-            " have enough capacity {warehouse_id = " + warehouseLoadDto.getId() + "}");
+            " have enough capacity {warehouse_id = " + id + "}");
     }
 
     @Override
-    public SavedItemDto outcomeItem(SavedItemDto savedItemDto, int quantity) {
+    public SavedItemDto outcomeItem(SavedItemDto savedItemDto, int quantity, User user, Associate associate) {
         int difference = savedItemDto.getQuantity() - quantity;
-        if (savedItemDao.findSavedItemByItemId(savedItemDto.getItemId()).getQuantity() >= quantity) {
-            savedItemDao.outComeSavedItem(savedItemDtoMapper.convertSavedItemDtoToSavedItem(savedItemDto), difference);
+        if (savedItemDao.findSavedItemByWarehouseId(savedItemDto.getWarehouseId()).getQuantity() >= quantity) {
+            savedItemDao.outComeSavedItem(savedItemDtoMapper.toEntity(savedItemDto), difference);
+            transactionDao.create(ItemTransactionUtil.createTransaction(savedItemDto, user, associate, TransactionType.OUT));
+            eventService.create(ItemEventUtil.createEvent(savedItemDto, user, EventName.ITEM_SHIPPED));
             savedItemDto.setQuantity(difference);
             return savedItemDto;
         }
+        eventService.create(ItemEventUtil.createEvent(savedItemDto, user, EventName.ITEM_ENDED));
```

Comment:
**IMPORTANT: In order to not block your PR please do this in next one!**
This event should be logged when there are less than 10 items left. (Move this to property file). So need to add additional condition